sale_coupon_multi_use/models/sale_coupon.py

```diff
@@ -19,11 +19,14 @@ class SaleCoupon(models.Model):
     # Takes value from related program (coupon_multi_use field), when
     # it is generated.
     multi_use = fields.Boolean(readonly=True)
+    currency_program_id = fields.Many2one(related="program_id.currency_id")
```

Comment:
To move in your `sale_coupon_multi_currency` module: it's not linked to multi use.